flow/project.py

```diff
@@ -267,6 +268,31 @@ def _make_bundles(operations, size=None):
             break
 
 
+class _HooksRegistry:
+    """Class storing hooks at the operation level for an instance of `:class:`~.FlowProject`."""
+
+    def __init__(self):
+        self._operation_hooks = defaultdict(lambda: defaultdict(list))
+
+    def __getitem__(self, func):
+        return self._operation_hooks[func]
+
+    def __getattr__(self, name):
+        if name in ("on_start", "on_finish", "on_success", "on_fail"):
+
+            class _InstallHook:
+                def __init__(install_self, hook_func):
+                    install_self.hook_func = hook_func
+
+                def __call__(install_self, func):
+                    self._operation_hooks[func][name].append(install_self.hook_func)
+                    return func
+
+            return _InstallHook
+        else:
+            raise AttributeError
```

Comment:
I think this needs to raise an instance of `AttributeError` with a message, not the class itself.